File: java/src/tk/csparpa/gofdp/adapter/Adaptee.java
package tk.csparpa.gofdp.adapter;

/**
 * This is the "legacy" class that must be adapted to the target interface.
 */
public class Adaptee {
	
	private double operand1;
	private double operand2;
	
	public Adaptee(double operand1, double operand2) {
		this.operand1 = operand1;
		this.operand2 = operand2;
	}

	public double getFirstOperand(){
		return this.operand1;
	}
	
	public double getSecondOperand(){
		return this.operand2;
	}
	
	public double computeSum(){
		return this.operand1 + this.operand2;
	}
	
	public double max(){
		return Math.max(this.operand1, this.operand2);
	}
}

File: python/adapter/adaptee.py
class Adaptee():
    """
    This is the "legacy" class that must be adapted to the target interface.
    """    
    _operand_1 = None
    _operand_2 = None
    
    def __init__(self, operand_1, operand_2):
        self._operand_1 = operand_1
        self._operand_2 = operand_2

    def get_first_operand(self):
        return self._operand_1
    
    def get_second_operand(self):
        return self._operand_2
    
    def compute_sum(self):
        return self._operand_1 + self._operand_2
    
    def max(self):
        return max(self._operand_1, self._operand_2)